ARCHITECTURE.md

```diff
@@ -188,3 +194,34 @@ kURL automation, more specifically the `import-external-addons` GitHub action, p
 New versions are published to the external add-on registry and packages are copied from the source and stored in the kURL S3 bucket.
 
 The kURL API merges the external add-on registry with its internal list of add-on versions, making them available to the end-user.
+
+## Deployment and Releases
+
+Upon releasing kURL, scripts and add-on package archives are built and uploaded to to the kURL object storage bucket along with metadata including the Git sha from which they were generated.
+Once complete, the VERSION file is updated to point to the current version of kURL.
+The API makes use of this VERSION file to resolve the scripts and add-on packages.
+
+### Production Workflow
+
+Production release are triggered by running the command `make tag-and-release` or pushing a tag in the format "v*.*.*".
+Production releases are uploaded to the object storage bucket at prefix `dist`, for example https://kurl-sh.s3.amazonaws.com/dist/v2022.04.19-0/.
+Before building add-on packages, the workflow will first check if there were any changes made to the add-on since the previous production release based on metadata included with the add-on package.
+If no changes were made, the package will be copied from the previous production release to optimize for build times.
+Next, the workflow will check if there were any changes made since the previous staging release and copy from staging if no changes were made.
+The is to account for a scenario where a commit were being tagged other than what is at the HEAD of main.
+If changes were made to both production and staging, the package will be built from source and uploaded.
+Finally, VERSION file is updated at https://kurl-sh.s3.amazonaws.com/dist/VERSION to point to the new production version.
+Historical production releases are never removed from the object storage bucket.
```

Comment:
So, we have like 4GB for each release and we are keeping them for each version releases. Is that?

Reply:
Have we not specific jobs that are trigged in this scenario?
What tests are executed in this scenario that are not executed when we push a PR?

Reply:
Yes, people intentionally run previous kURL versions - it's a way for vendors to do their own QA for a release before inflicting it upon their users

Reply:
mainly answered .. 
I have just one point here that I will ask offiline